userdocs/diagnostics/multiple-inheritance.md

```diff
@@ -0,0 +1,51 @@
+# Multiple Inheritance
+
+In some programming languages, a class can inherit the interface of multiple base classes. Known as multiple inheritance, this feature can add significant complexity to the language and is unsupported in Swift. Instead, Swift allows composition of interfaces using protocols.
+
+Consider the following example:
+
+```swift
+protocol Utensil { 
+    var name: String { get }
+} 
+
+/* ------------------------------ Serving Utensil ------------------------------ */
+protocol ServingUtensil: Utensil {
+    func serve()
+} 
+
+extension ServingUtensil {
+    func serve() { /* ... */ }
```

Comment:
```suggestion
    func serve() { /* Default implementation. */ }
```

We can make this the one example of a default implementation and emphasize that point by using a comment.